src/deluge/gui/views/instrument_clip_view.cpp

```diff
@@ -3844,26 +3844,51 @@ void InstrumentClipView::drawNoteCode(uint8_t yDisplay) {
 }
 
 void InstrumentClipView::drawDrumName(Drum* drum, bool justPopUp) {
+	char drumName[50];
 
-	char const* newText;
+	getDrumName(drum, drumName);
 
 	if (display->haveOLED()) {
-		char buffer[50];
+		display->popupText(drumName);
+	}
+	else {
+		bool andAHalf;
+		if (drum && (drum->type == DrumType::SOUND)
+		    && (display->getEncodedPosFromLeft(99999, drumName, &andAHalf) > kNumericDisplayLength)) {
+			display->setScrollingText(drumName, 0, kInitialFlashTime + kFlashTime);
+		}
+		else {
+			if (justPopUp && currentUIMode != UI_MODE_AUDITIONING) {
+				display->displayPopup(drumName);
+			}
+			else {
+				display->setText(drumName, false, 255, true);
+			}
+		}
+	}
+
+	if (drum && drum->type != DrumType::SOUND) {
+		if (drum->type == DrumType::MIDI) {
+			indicator_leds::blinkLed(IndicatorLED::MIDI, 1, 1);
+		}
+		else if (drum->type == DrumType::GATE) {
+			indicator_leds::blinkLed(IndicatorLED::CV, 1, 1);
+		}
+	}
+}
 
+void InstrumentClipView::getDrumName(Drum* drum, char* drumName) {
```

Comment:
This might benefit from taking a `StringBuf&` rather than a `char*`, since we could use the checked functions from that class.

Reply:
> This might benefit from taking a `StringBuf&` rather than a `char*`, since we could use the checked functions from that class.

Changed it to StringBuf, can you let me know if that's what you meant? thx